ibis/pyspark/compiler.py

```diff
@@ -1011,3 +1058,294 @@ def compile_cumulative_max(t, expr, scope, *, window, **kwargs):
     return _handle_cumulative_operation(
         t, expr, scope, F.max, window=window, **kwargs
     )
+
+
+# -------------------------- Temporal Operations ----------------------------
+
+# Ibis value to PySpark value
+_time_unit_mapping = {
+    'Y': 'year',
+    'Q': 'quarter',
+    'M': 'month',
+    'W': 'week',
+    'D': 'day',
+    'h': 'hour',
+    'm': 'minute',
+    's': 'second'
+}
+
+
+@compiles(ops.Date)
+def compile_date(t, expr, scope, **kwargs):
+    op = expr.op()
+    src_column = t.translate(op.arg, scope)
+    return F.to_date(src_column).cast('timestamp')
+
+
+def _extract_x_from_datetime(t, expr, scope, extract_fn, **kwargs):
```

Comment:
Can we rename this to `_extract_component_from_datetime`?

Reply:
Yep, done.